common/esxi_get_guest_config_options.yml

```diff
@@ -0,0 +1,286 @@
+        | map('lower')
+        | map("replace", " ", "_")
+        | zip(get_config_options_result.instance["Recommended config options"].values())
+        | items2dict(key_name=0, value_name=1)
+      }}
+  when:
+    - get_config_options_result is defined
+    - get_config_options_result.instance is defined
+    - get_config_options_result.instance["Recommended config options"] is defined
+
+# Get config options from config file
+- block:
+    - name: "Initialize variables for getting guest ID default configs"
+      set_fact:
+        vm_config_option_esx_hw: "/etc/vmware/hostd/env/vmconfigoption-esx-hw{{ esxi_hardware_version }}.xml"
+        guest_config_options_xpath: "/ConfigRoot/ConfigurationOption/guestOSDescriptor/e[id='{{ guest_id }}']"
+        vm_device_types:
+          vim.vm.device.VirtualLsiLogicController: 'lsilogic'
+          vim.vm.device.VirtualLsiLogicSASController: 'lsilogicsas'
+          vim.vm.device.ParaVirtualSCSIController: 'paravirtual'
+          vim.vm.device.VirtualAHCIController: 'sata'
+          vim.vm.device.VirtualNVMEController: 'nvme'
+          vim.vm.device.VirtualIDEController: 'ide'
+          vim.vm.device.VirtualVmxnet3: 'vmxnet3'
+          vim.vm.device.VirtualE1000e: 'e1000e'
+          vim.vm.device.VirtualE1000: 'e1000'
+          vim.vm.device.VirtualVmxnet3Vrdma: 'pvrdma'
+          vim.vm.device.VirtualSriovEthernetCard: 'sriov'
+          vim.vm.device.VirtualPCNet32: 'pcnet32'
+          vim.vm.device.VirtualVmxnet2: 'vmxnet2'
+          vim.vm.device.VirtualUSBController: 'usb2'
+          vim.vm.device.VirtualUSBXHCIController: 'usb3'
+        vm_config_name:
+          recommendedCdromController: "default_cdrom_controller"
+          numRecommendedCoresPerSocket: "default_cpu_cores_per_socket"
+          numRecommendedPhysicalSockets: "default_cpu_socket"
+          recommendedUSBController: "default_usb_controller"
+          recommendedDiskController: "default_disk_controller"
+          recommendedDiskSizeMB: "default_disk_size_in_mb"
+          recommendedFirmware: "default_firmware"
+          recommendedMemMB: "default_memory_in_mb"
+          recommendedEthernetCard: "default_network_adapter"
+          defaultSecureBoot: "default_secure_boot"
+          supportsSecureBoot: "support_secure_boot"
+          fullName: "guest_fullname"
+          recommendedPersistentMemoryMB: "rec_persistent_memory"
+          supportedMinPersistentMemoryMB: "support_min_persistent_mem_mb"
+          persistentMemorySupported: "support_persistent_memory"
+          supportsTPM20: "support_tpm_20"
+
+    - name: "Initialize VM default config options"
+      set_fact:
+        guest_config_options: >-
+          {{
+            vm_config_name.values() | zip_longest([]) 
+            | items2dict(key_name=0, value_name=1)
+          }}
+
+    # Create a temp file to store config options
+    - include_tasks: create_temp_file_dir.yml
```

Comment:
There is a cache folder `local_cache` created at the beginning for downloaded files.

Reply:
Thanks! I just added tmp_dir variable for this task.